app/models/wds-on-site-notifications/wds-on-site-notification.js

```diff
@@ -77,9 +78,29 @@ export default EmberObject.extend({
 				return notificationTypes.articleCommentAtMention;
 			case 'article-comment-reply-at-mention-notification':
 				return notificationTypes.articleCommentReplyAtMention;
+			case 'message-wall-post-notification':
+				return notificationTypes.messageWallPost;
+			case 'message-wall-reply-notification':
+				return notificationTypes.messageWallReply;
+			case 'wall-post-removed-notification':
+				return notificationTypes.messageWallPostRemoved;
 		}
 	},
 
+	getMetadata(notification) {
+		let metadata =
+			typeof notification.metadata === 'undefined' || notification.metadata === null ? null : notification.metadata;
+		if (typeof metadata === 'string') {
+			try {
+				metadata = JSON.parse(metadata);
+			} catch (e) {
+				// continue regardless of error
+			}
+		}
+		// Caveat: if the metadata was not serialized JSON, a string could be returned instead of an object
+		return metadata;
```

Comment:
Is it handled later as string and object?

Reply:
It depends on the notification type. The metadata is 100% developer controlled as to whether it's a string or object for the notification they're creating.